ptf_tests/common/utils/test_utils.py

```diff
@@ -1021,3 +1041,123 @@ def host_check_netperf(remote=False, hostname="",username="",password=""):
+        eth_dst='\"FF:FF:FF:FF:FF\"'
+    
+    #packet build    
+    command_list = ['python3','import scapy.all as scapy']
+    
+    for cmd in command_list:
+        status = conn.sendCmd(cmd)
+        if status:
+            print(f"Command: {cmd} executed ")
+        else:
+            print("Failed to execute command {cmd}")
+    
+    #starting log on remote vmcmd="open(\"%s\").read()" %pcap_file
+    print(f"capturing in to pcap file {pcap_file} on remote_host")
+    tcpdump_cmd = "tcpdump -i %s -w %s" % (iface, pcap_file)
+    #remote_conn.sendCmd('tcpdump -i ens3 -w /tmp/log_cap1.pcap &')
+    remote_conn.sendCmd(tcpdump_cmd)
+
+    # Packet Building using scapy
+    conn.sendCmd('pkt=scapy.Ether(dst=%s,src=%s)/scapy.IP(dst=%s,src=%s)/\
+                  scapy.TCP()/scapy.Raw(\'0\'*50)' %(eth_dst,eth_src,ip_dst,ip_src))
+
+    print('scapy.sendp(pkt,iface=\"%s\",count=%s)' % (iface,num))
+    status = conn.sendCmd('scapy.sendp(pkt,iface=\"%s\",count=%s)' % (iface,num))
+    # status = conn.sendCmd(cmd)
+    if status:
+        print(f"Command: scapy send executed ")
+    else:
+        print("Failed to execute command scapy send")
+
+    result = conn.readResult()
+    if result:
+        match = re.search('Sent (\d*) packets', result)
+        if match:
+            pkt_sent = int(match.group(1))
+            print("match.group(0)")
+  
+    if f"Sent {num} packets" in result:
+        print(f"PASS: Sent {num} to dst_ip: {ip_dst}")
+        print(f"result:{result}")
+        return True
+    else:
+        print(f"FAIL: Sent Failed--> {result} sent, expected {num} packets to be sent to dst_ip: {ip_dst}")
+        return False
+
+
+def verify_scapy_traffic_from_vm(vm_id,conn,config_data):
+
+    result = False
+    packets = []
+    count = config_data['traffic']['number_pkts'][0]
+    pcap_file = config_data['traffic']['pcap_file_name'][0]
+
+    send_ctrl_c(conn)
+  
+    cmd_list = ['python3', "from scapy.all import *", 
+                'ls %s' % pcap_file]
+
+    for cmd in cmd_list:
+        conn.sendCmd(cmd)
+    if conn.readResult():
+        print(f"Pcap file {pcap_file} exists")
+    else:
+        print("Pcap file {pcap_file} does not exist or empty")
+
+    cmd_list = ['python3', "from scapy.all import *",
+                'packets=[x for x in sniff(offline=\"%s\")]' %pcap_file]
+
+    for cmd in cmd_list:
+        conn.sendCmd(cmd)
+
+    conn.sendCmd('out_pkts= [1 for pkt in packets if pkt.version==4]')
+    conn.sendCmd('len(out_pkts)')
+    out1 = conn.readResult()
+    out= out1.split('\n')
+    if out:
+        match = re.search('.*(\d+).*', out[-2])
+        if match:
+            pkt_count = int(match.group(0)[:-1])
+
+    print(f"len of pkts {pkt_count}")
+
+    if pkt_count ==  count:
+        result = True
+        print(f"Received {pkt_count} on vm_id {vm_id}")
+    else:
+        print(f"FAIL: {pkt_count} received on vm_id {vm_id}, expected: {count}")
+
+    return result
+
```

Comment:
General comment: Since you are running many commands in the python shell, do you feel creating a .py script in the test_utils.py and then executing that script would be cleaner?

Reply:
I have thought through creation of python file  as  second approach. I would prefer to merge this code and rework.